Documentation/Fr/Index.xyl

```diff
@@ -0,0 +1,732 @@
+  trouve notre problématique.</p>
+
+  <h3 id="Test_test_case_and_test_suite" for="main_toc">Test, scénario de test et
+  suite de test</h3>
+
+  <p>Officiellement, un <strong>test</strong> d'un SUT est un ensemble de données
+  (appelées données de test) qui alimentent les valeurs des arguments (aussi
+  appelés entrées) d'un SUT. Un <strong>scénario de test</strong> est une paire
+  composée d'un état (le contexte dans lequel est exécuté le test) et un test.
+  Une <strong>suite de test</strong> est ensemble de scénario de test. Par
+  conséquent, il y a deux états du point du vue du test : un
+  <strong>pré-état</strong> — avant l'exécution du SUT — et un
+  <strong>post-état</strong> — après l'exécution —.</p>
+  <p>Construire le pré-état est critique : le but est de mettre le SUT dans un
+  (pré-)état spécifique afin de le tester. La partie du scénario de test responsable
+  de créer le pré-état est appelé le <strong>préambule</strong>. Ainsi, le
+  préambler est le code qui met le SUT dans un état spécifique.</p>
+  <p>Ainsi, en écrivant un scénario de test, nous aurons 3 éléments à considérer :</p>
+  <ol>
+    <li>Le <strong>préambule</strong>, pour mettre le SUT dans un état spécifique ;</li>
+    <li>L'<strong>exécution</strong>, pour exécuter le SUT et obtenir un résultat ;</li>
+    <li>L'<strong>oracle</strong>, pour évaluer le verdict (basé sur le résultat
+    de l'exécution et le post-état).</li>
+  </ol>
+  <p>L'oracle consiste en une séquence d'assertions : “Est-ce que cette valeur
+  est égale à… ?”, “Est-ce que la valeur a changé de cette façon… ?” etc. Il
+  est censé pouvoir vérifier la forme de certaines données mais également
+  comparer des données entre le pré et le post-état.</p>
+  <p>Dans un scénario de test, nous représentons et introduisons le préambule,
+  l'exécution et l'oracle respectivement par les mots clés <code>given</code>,
+  <code>when</code> et <code>then</code>. Ceci sera détaillé ci-après.</p>
+
+  <h3 id="Test_framework" for="main-toc">Framework de test</h3>
+
+  <p>Hoa n'a pas développé son propre framework de tests pour plusieurs raisons.
+  La raison principale étant que c'est une tâche laborieuse. Cela implique plus
+  de développement, plus d'outils à maintenir, moins de gens concentrés sur Hoa
+  etc. A la place, nous préférons choisir un bon projet sur lequel nous
+  appuyer et y contribuer. Nous avons opté pour <a href="http://atoum.org/">atoum</a>,
+  <q>“un framework de test simple, moderne et intuitif”</q>.</p>
+  <p>Ainsi, la bibliothèque <code>Hoa\Test</code> est la base pour tester toutes
+  les bibliothèques de Hoa. Elle intègre atoum et vise à être un pont entre
+  les fonctionnalités de Hoa et atoum. Elle ajoute également quelques
+  fonctionnalités si nécessaire. Elle est plus qu'une couche d'abstraction au
+  dessus d'atoum. <code>Hoa\Test</code> fournit également sa propre ligne de
+  commande pour exécuter les tests avec everything pre-packaged. En conséquence,
```

Comment:
@Hywan "everything pre-packaged" isn't clear for me (I don't understand what is "everything").